src/test/java/org/jabref/logic/bibtexkeypattern/BibtexKeyGeneratorTest.java

```diff
@@ -942,22 +942,22 @@ void crossrefkeywordsNKeywordsSeparatedBySpace() {
 
     @Test
     void testCheckLegalKeyEnforceLegal() {
-        assertEquals("AAAA", BibtexKeyGenerator.cleanKey("AA AA", true));
-        assertEquals("SPECIALCHARS", BibtexKeyGenerator.cleanKey("SPECIAL CHARS#{\\\"}~,^", true));
-        assertEquals("", BibtexKeyGenerator.cleanKey("\n\t\r", true));
+        assertEquals("AAAA", BibtexKeyGenerator.cleanKey("AA AA", "-`ʹ:!;?^+"));
+        assertEquals("SPECIALCHARS", BibtexKeyGenerator.cleanKey("SPECIAL CHARS#{\\\"}~,", "-`ʹ:!;?^+"));
+        assertEquals("", BibtexKeyGenerator.cleanKey("\n\t\r", "-`ʹ:!;?^+"));
     }
 
     @Test
     void testCheckLegalKeyDoNotEnforceLegal() {
-        assertEquals("AAAA", BibtexKeyGenerator.cleanKey("AA AA", false));
-        assertEquals("SPECIALCHARS#~^", BibtexKeyGenerator.cleanKey("SPECIAL CHARS#{\\\"}~,^", false));
-        assertEquals("", BibtexKeyGenerator.cleanKey("\n\t\r", false));
+        assertEquals("AAAA", BibtexKeyGenerator.cleanKey("AA AA", "-`ʹ:!;?^+"));
+        assertEquals("SPECIALCHARS", BibtexKeyGenerator.cleanKey("SPECIAL CHARS#{\\\"}~,^", "-`ʹ:!;?^+"));
```

Comment:
I think it's better to use an empty string as the `unwanted` characters here, instead of changing the expected value. Otherwise this test is exactly the same as the previous.

Reply:
Of course, I don't know why I didn't pay attention for this. Thank you very much for your helpful review!